spec/rspec/expectations/handler_spec.rb

```diff
@@ -130,6 +130,18 @@ module Expectations
 
           RSpec::Expectations::PositiveExpectationHandler.handle_matcher(actual, matcher, failure_message)
         end
+
+        it "add custom failure message to mock matcher if there is one" do
+          error_generator = double(:opts => { :message => {} })
+          verifying_message_expectation = double(:error_generator => error_generator)
```

Comment:
Whats the real thing for this, can we either use a real one or use an `instance_double`?

Reply:
Ah it's mock specific isn't it, should be able to use an instance double of `RSpec::Mocks::ErrorGenerator` etc?

Reply:
I was not sure about this when we wrote the test. Do we want to see some `rspec-mocks` code into our specs? We have only one occurrence at the moment in `rspec-expectation`.

https://github.com/rspec/rspec-expectations/blob/ed736c3fc996ca0c978e869806bf41359db395a3/spec/rspec/expectations/configuration_spec.rb#L32

Reply:
This is integration with `rspec-mocks` so I think it's important to double the right thing. This way if the implementation changes on the other side it will fail. You can add a note as well explaining if that makes you feel more comfortable?

Reply:
I prefer with `instance_double`. I did the change in https://github.com/rspec/rspec-expectations/pull/1156/commits/7a776d6aa8f71dff8186dda33da2227dc4a15a39. @JonRowe if you can just confirm it's ok to have `RSpec::Mocks` references in `rspec-expectation` then we are good. 😊